vortex-array/src/builders/fixed_size_list.rs

```diff
@@ -181,15 +180,14 @@ impl ArrayBuilder for FixedSizeListBuilder {
     /// - If elements are nullable, we use null values
     /// - If elements are non-nullable, we use zero values
     fn append_nulls(&mut self, n: usize) {
-        let element_count = n * self.list_size() as usize;
+        assert!(
+            self.dtype.is_nullable(),
+            "tried to append {n} nulls to a non-nullable array builder"
+        );
```

Comment:
Shall we have a public `append_nulls` that call a `_append_nulls()` and do validation in `append_nulls`?